frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,210 @@
+import * as React from 'react';
+import { Alert } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { LoadingBox, PageHeading } from '@console/internal/components/utils';
+import { PodKind, k8sCreate, k8sKillByName } from '@console/internal/module/k8s';
+import { PodExecLoader } from '@console/internal/components/pod';
+import { PodModel } from '@console/internal/models';
+import {
+  useK8sWatchResource,
+  WatchK8sResult,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { resourcePath } from './utils/resource-link';
+import { getBreadcrumbPath } from '@console/internal/components/utils/breadcrumbs';
+
+export const isContainerCrashLoopBackOff = (pod: PodKind, containerName: string): boolean => {
+  const containerStatus = pod?.status?.containerStatuses?.find((c) => c.name === containerName);
+  const waitingReason = containerStatus?.state?.waiting?.reason;
+  return waitingReason === 'CrashLoopBackOff';
+};
+
+export const isWindowsPod = (pod: PodKind): boolean => {
```

Comment:
We have similar logic in pod-exec.jsx. I'd suggest defining this in one place and calling it from both places.